pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -111,6 +112,19 @@ func addLoggerAnnotations(logger *v1alpha2.Logger, annotations map[string]string
 	return false
 }
 
+func addLoggerContainerPort(container *v1.Container) {
+	if container != nil {
```

Comment:
Shouldn't the container always ne non nil. It should be an error if not?

Reply:
Yes you are right, I can remove this check if you think it's unneccessary.